hips/hip-9999.md

```diff
@@ -0,0 +1,32 @@
+---
+hip: 9999
+title: "Helm Resolving Dependencies First Option"
+authors: [ "protosam" ]
+created: "2021-09-27"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+All chart dependencies are rendered with the parent chart before being sent to the cluster. This can prevent parent charts from using CRDs from charts that manage their own CRD life cycle. Resolving and installing dependencies in a specified order can solve this issue.
```

Comment:
> This can prevent parent charts from using CRDs from charts that manage their own CRD life cycle.

Can you elaborate further on this point? More specifically, why does rendering all charts at once prevent this?

It is a good idea to point to existing documentation like https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#install-a-crd-declaration-before-using-the-resource when presenting your case. That way the reader has a better understanding of the issues involved. Though I would suggest going into more detail under "Motivation", since that section is where you should explain why the existing design is inadequate to address the problem.